### ISSUE TITLE
[9/9] Meta-labeling: learn when signals are profitable

### ISSUE THREAD

**Issue by kumargu** [kumargu/OpenQuant#89]
## Priority: 🟢 Execute NINTH (3-4 hrs)
**Epic:** #80 | **Research:** #79 | **Depends on:** #82 (regime), #85 (GARCH)

## Why Meta-Labeling

Your 4 strategies generate signals. Some signals are good, some are noise. Currently you have no way to tell which is which at signal time. **Meta-labeling trains a classifier to predict: "given this signal, will the trade be profitable?"**

The meta-model's output (probability) directly becomes the position size scalar: high confidence = full size, low confidence = small or skip.

## The Architecture

```
Primary models (existing strategies):
  Mean Reversion → Buy/Sell signal
  Momentum       → Buy/Sell signal    ──→ Meta-model ──→ P(profitable) ──→ Position size
  Breakout       → Buy/Sell signal           ↑
  VWAP Reversion → Buy/Sell signal     Meta-features:
                                        - strategy scores
                                        - regime (from BOCPD)
                                        - GARCH vol
                                        - volume profile
                                        - time of day
                                        - cross-strategy agreement
```

### 1. Generate training labels via Triple Barrier Method

```python
# In paper_trading/ (new file: meta_labeling.py)

def triple_barrier_label(
    entry_price, future_prices, 
    take_profit_pct=0.02,  # 2% TP
    stop_loss_pct=0.015,   # 1.5% SL  
    max_hold_bars=50,      # time barrier
):
    """
    Label each trade signal as 1 (profitable) or 0 (not).
    Whichever barrier is hit first determines the label.
    """
    for i, price in enumerate(future_prices):
        ret = (price - entry_price) / entry_price
        if ret >= take_profit_pct:
            return 1  # hit take-profit → profitable
        if ret <= -stop_loss_pct:
            return 0  # hit stop-loss → not profitable
        if i >= max_hold_bars:
            return 1 if ret > 0 else 0  # time barrier → label by sign
    return 0
```

### 2. Build meta-features from engine output

For each signal that fired, collect:
```python
meta_features = {
    'strategy_score': signal.score,
    'strategy_name': signal.reason,  # one-hot encode
    'z_score': signal.z_score,
    'relative_volume': signal.relative_volume,
    'garch_vol': features.garch_vol,
    'market_regime': features.market_regime,  # from BOCPD
    'regime_change_prob': features.regime_change_probability,
    'hour_of_day': bar.timestamp.hour,
    'day_of_week': bar.timestamp.weekday(),
    'n_strategies_agree': combiner.agreement_count,
    'rolling_sharpe_5d': strategy.rolling_sharpe(5),
    'rolling_win_rate_20': strategy.rolling_win_rate(20),
    'adx': features.adx,
    'atr_ratio': features.atr / price,
}
```

### 3. Train a gradient-boosted classifier

```python
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.calibration import CalibratedClassifierCV

# Use purged CV from issue #87!
model = CalibratedClassifierCV(
    GradientBoostingClassifier(n_estimators=100, max_depth=3),
    cv=purged_kfold_cv(n_splits=5),
    method='isotonic'  # calibrated probabilities
)
model.fit(X_train, y_train)

# Export for Rust inference (simple decision rules or ONNX)
```

### 4. Use meta-model output for sizing

The meta-model outputs P(profitable) ∈ [0, 1]:
```rust
// In risk::check() or combiner:
let meta_confidence = meta_model.predict(meta_features);

// Only trade if meta-model confidence > threshold
if meta_confidence < 0.5 {
    return None; // skip this signal
}

// Scale position by confidence
let size_scalar = (meta_confidence - 0.5) * 2.0; // maps [0.5, 1.0] → [0.0, 1.0]
let qty = kelly_qty * size_scalar;
```

### 5. Offline training, online inference

- **Train offline** in Python using historical signals + labels (nightly or weekly)
- **Export** simple decision tree rules or ONNX model
- **Infer online** in Rust (decision tree is trivially fast; or call Python sidecar)

## Definition of Done
- [ ] Triple Barrier labeling for historical signals
- [ ] Meta-feature extraction from engine output
- [ ] Gradient-boosted meta-classifier with calibrated probabilities
- [ ] Purged CV used for training (from #87)
- [ ] Meta-confidence threshold for trade gating
- [ ] Position scaling by meta-confidence
- [ ] Offline training pipeline (Python)
- [ ] Online inference path (Rust or Python sidecar)

## References
- De Prado (2018) *Advances in Financial Machine Learning*, Ch. 3 (meta-labeling), Ch. 50 (triple barrier)
- [Hudson & Thames meta-labeling](https://github.com/hudson-and-thames/meta-labeling)
- [Does Meta Labeling Add to Signal Efficacy?](https://hudsonthames.org/does-meta-labeling-add-to-signal-efficacy-triple-barrier-method/)
- [mlfinlab triple barrier](https://github.com/FGU1/mlfinlab)
